akka-testkit-typed/src/test/scala/akka/testkit/typed/BehaviorTestkitSpec.scala

```diff
@@ -130,4 +130,12 @@ class BehaviorTestkitSpec extends WordSpec with Matchers {
       effects shouldBe Seq(SpawnedAdapter)
     }
   }
+
+  "BehaviorTestkit's run" can {
+    "run behaviors with messages without canonicalization" in {
+      val testkit = BehaviorTestkit[Father.Command](Father.init())
+      testkit.runUncanonical(SpawnAdapterWithName("adapter"))
+      testkit.currentBehavior shouldBe Behavior.same
```

Comment:
and what happens if you `run` one more time after this?

hm, I think I understand the point. Might be somewhat confusing to return the behavior and not update `current`, if that is what you mean. How do you proceed with next message. Perhaps we could simply keep the previous un-canonicalized behavior around and accessible with `returnedBehavior(): Behavior` so that you after `run()` can assert on that, but the `current` is always the canonicalized as it is now.

Reply:
That sounds actually nice :)
I'll update the code